rootless/specconv/specconv_linux.go

```diff
@@ -26,10 +27,19 @@ func getCurrentOOMScoreAdj() int {
 	return i
 }
 
-func toRootless(spec *specs.Spec, currentOOMScoreAdj int) error {
-	// Remove cgroup settings.
-	spec.Linux.Resources = nil
-	spec.Linux.CgroupsPath = ""
+func toRootless(spec *specs.Spec, noCgroup bool, currentOOMScoreAdj int) error {
+	if noCgroup {
+		// Remove cgroup settings.
+		spec.Linux.Resources = nil
+		spec.Linux.CgroupsPath = ""
+	} else {
+		if spec.Linux.Resources != nil {
+			// Remove devices: https://github.com/containers/crun/issues/255
+			spec.Linux.Resources.Devices = nil
+			// Remove blockIO as a workaround of https://github.com/containers/crun/issues/254
+			spec.Linux.Resources.BlockIO = nil
```

Comment:
Maybe we should parse cgroup.subtree_control (via systemd API, if exists?) and do this conditionally.